generators/csharp/model/src/union/UnionSerializationTestGenerator.ts

```diff
@@ -0,0 +1,128 @@
+import { CSharpFile, FileGenerator, csharp } from "@fern-api/csharp-codegen";
+import { RelativeFilePath, join } from "@fern-api/fs-utils";
+
+import { TypeDeclaration } from "@fern-fern/ir-sdk/api";
+
+import { ModelCustomConfigSchema } from "../ModelCustomConfig";
+import { ModelGeneratorContext } from "../ModelGeneratorContext";
+
+import TestInput = TestClass.TestInput;
+
+export declare namespace TestClass {
+    interface TestInput {
+        objectInstantiationSnippet: csharp.CodeBlock;
+        json: unknown;
+    }
+}
+
+const SERIALIZATION_TEST_FOLDER = RelativeFilePath.of("Unit/Serialization");
+
+export class UnionSerializationTestGenerator extends FileGenerator<
```

Comment:
Awesome that we included this upfront here 👏

Reply:
Was kinda crucial to really testing and debugging it. It's a breeze to debug it now.